src/server/nodeimpl.ts

```diff
@@ -0,0 +1,7 @@
+/// <reference path="node.d.ts" />
+module ts.server {
+    export class NodeEnvironment implements Environment {
+        byteLength = Buffer.byteLength;

```

Comment:
these are only used in the session object, i do not see the point of the additional abstraction, just pass them as constructor paramters.